### PR TITLE
JAMES-3867 Make IMAP modular

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/james-project/issues/4690

jira: TODO

### CODE REVIEW

**Review by chibenwa — COMMENTED** [review]
ImapHandlerPackage needs to be loaded from the configuration (`imapServer.xml`)

File: server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/ImapPackage.java [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+        };
+    }
+
+    ImapPackage DEFAULT = and(ImmutableList.of(new DefaultNoAuth(), new DefaultAuth()));
+
+    class DefaultNoAuth extends Impl {
+        public DefaultNoAuth() {
+            super(
+                ImmutableList.of(
+                    CapabilityProcessor.class,
```
Comment (chibenwa) [inline server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/ImapPackage.java:185]: If we are rebased on master we need to add here IdProcessor + its encoder and command parser

File: server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/ImapPackage.java [hunk 2]
```diff
@@ -0,0 +1,314 @@
+package org.apache.james.modules.protocols;
```
Comment (chibenwa) [inline server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/ImapPackage.java:1]: Missing license

File: server/container/guice/common/src/main/java/org/apache/james/TemporaryJamesServer.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.server.core.configuration.Configuration;
+
+import com.google.common.collect.ImmutableList;
+
+public class TemporaryJamesServer {
```
Comment (chibenwa) [inline server/container/guice/common/src/main/java/org/apache/james/TemporaryJamesServer.java:38]: Maybe not needed. Can't we just use a MemoryJamesServer for this test?

File: server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/ImapPackage.java (hunk 1 above)
Comment (vttranlina) [reply server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/ImapPackage.java:185]: `org.apache.james.imap.processor.IdProcessor` already
yes, i forgot encoder/decoder

**Comment by chibenwa** [comment]
<p><span style="font-weight:bold">org.apache.james.imap.processor.GetAnnotationProcessorTest.processShouldGetAnnotationsAndReturnCompleteResponseWithTruncateDataByLessThenOrEqualMaxsize</span></p><div style="text-align:right;">Failing for the past 1 build
          (Since <a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1343/5/" class="model-link"><span style="width: 16px; height: 16px; " class="build-status-icon__wrapper icon-red icon-sm"><span class="build-status-icon__outer"><svg viewBox="0 0 24 24" focusable="false" class="svg-icon "><use href="https://ci-builds.apache.org/images/build-status/build-status-sprite.svg#build-status-static"></use></svg></span><svg viewBox="0 0 24 24" focusable="false" class="svg-icon icon-red icon-sm"><use href="https://ci-builds.apache.org/static/919f2990/images/build-status/build-status-sprite.svg#last-failed"></use></svg></span><span class="jenkins-icon-adjacent">#5</span></a> )
        </div><div style="text-align:right;"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1343/5/testReport/junit/org.apache.james.imap.processor/GetAnnotationProcessorTest/processShouldGetAnnotationsAndReturnCompleteResponseWithTruncateDataByLessThenOrEqualMaxsize/history">Took 4 ms.</a></div><div id="description"><div></div><div class="jenkins-buttons-row jenkins-buttons-row--invert"></div></div>
<h3>Stacktrace</h3><pre>java.lang.NullPointerException
	at org.apache.james.imap.processor.GetAnnotationProcessorTest.setUp(GetAnnotationProcessorTest.java:127)
</pre>org.apache.james.imap.processor.GetAnnotationProcessorTest.processShouldGetAnnotationsAndReturnCompleteResponseWithTruncateDataByLessThenOrEqualMaxsize
Failing for the past 1 build (Since
[#5](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1343/5/) )
[Took 4 ms.](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1343/5/testReport/junit/org.apache.james.imap.processor/GetAnnotationProcessorTest/processShouldGetAnnotationsAndReturnCompleteResponseWithTruncateDataByLessThenOrEqualMaxsize/history)
Stacktrace

java.lang.NullPointerException
	at org.apache.james.imap.processor.GetAnnotationProcessorTest.setUp(GetAnnotationProcessorTest.java:127)

Looks related...

**Pull request merged**